### PR TITLE
[extension]  2/n functionality for single turn prompting

### PR DESCRIPTION
**Pull request by Ankush-lastmile**
[extension]  2/n functionality for single turn prompting

Built out the conversational model parser for remote inference.

HF is in the process of revamping their libraries. Default behaviour of Converstaional is broken

Use the temporary fix is reccomended by HF https://github.com/huggingface/huggingface_hub/issues/2023#issuecomment-1935760040

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/1228).
* #1229
* __->__ #1228
* #1227

### CODE REVIEW

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py [hunk 1]
```diff
@@ -28,43 +28,37 @@
...
+    if "model" in completion_data:
+        # HF is in the process of revamping their libraries. Default behaviour of Converstaional is broken
+        # This temporary fix is reccomended by HF https://github.com/huggingface/huggingface_hub/issues/2023#issuecomment-1935760040
+        completion_data["model"] = (
+            f"https://api-inference.huggingface.co/pipeline/conversational/{completion_data['model']}"
+        )
+    else:
+        # manually set default model because, otherwise this will fail. see comment above.
+        # see this thread for more info: https://github.com/huggingface/huggingface_hub/issues/2023
+        completion_data["model"] = "https://api-inference.huggingface.co/models/facebook/blenderbot-400M-distill"
```
Comment (rossdanlm) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py:55]: nit: Just want to sanity check that this works for free so will be good for most people

**Review by rossdanlm — APPROVED** [review]

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py [hunk 2]
```diff
@@ -256,7 +250,7 @@ async def run_inference(
 
         response = client.translation(**completion_data)
```
Comment (rossdanlm) [inline extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py:251]: Don't we also need to update this to `client.conversational`?

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py [hunk 3]
```diff
@@ -28,43 +28,37 @@
...
-            "output_type": "execute_result",
-            "data": response,
-            "execution_count": 0,
-            "metadata": metadata,
-        }
-    )
-    return output
+    if "model" in completion_data:
+        # HF is in the process of revamping their libraries. Default behaviour of Converstaional is broken
+        # This temporary fix is reccomended by HF https://github.com/huggingface/huggingface_hub/issues/2023#issuecomment-1935760040
```
[the remark this answers is not in the document]
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py:48]: created #1232

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py (hunk 1 above)
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py:55]: Unfortunately I am unable to thoroughly test this locally due to unknown rate limiting issues. Will fix forward anything that comes up

File: extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py (hunk 2 above)
Comment (Ankush-lastmile) [reply extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/conversational.py]: updated, mistakenly kept it while testing.

**Comment by Ankush-lastmile** [comment]
translation -> conversational

**Pull request merged**